### PR TITLE
add exact match test to CI

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: add exact match test to CI

**Motivation and Context**
to improve CI pipeline stability

### CODE REVIEW

File: orttraining/tools/ci_test/download_e2e_test_data.py [hunk 1]
```diff
@@ -20,8 +20,8 @@
 import get_azcopy  # noqa: E402
 
 # update these if the E2E test data changes
-ARCHIVE_BLOB_URL = "https://onnxruntimetestdata.blob.core.windows.net/training/onnxruntime_training_data.zip?snapshot=2020-06-15T23:17:35.8314853Z"
-ARCHIVE_SHA256_DIGEST = "B01C169B6550D1A0A6F1B4E2F34AE2A8714B52DBB70AC04DA85D371F691BDFF9"
+# ARCHIVE_BLOB_URL = "https://onnxruntimetestdata.blob.core.windows.net/training/onnxruntime_training_data.zip?snapshot=2020-06-15T23:17:35.8314853Z"
+# ARCHIVE_SHA256_DIGEST = "B01C169B6550D1A0A6F1B4E2F34AE2A8714B52DBB70AC04DA85D371F691BDFF9"
```
Comment (thiagocrepaldi) [inline orttraining/tools/ci_test/download_e2e_test_data.py:25]:
```suggestion
```

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1196,12 +1204,13 @@ def run_training_python_frontend_e2e_tests(cwd):
...
-        run_subprocess([
-            'mpirun', '-n', str(ngpus), '-x', 'NCCL_DEBUG=INFO', sys.executable,
-            bert_pretrain_script], cwd=cwd)
+        # skip long run - it is very difficult to search for breaking commit with this long run test.
+        # # a long run
+        # log.debug('RUN: mpirun -n {} ''-x' 'NCCL_DEBUG=INFO'' {} {}'.format(
+        #     ngpus, sys.executable, bert_pretrain_script))
+        # run_subprocess([
+        #     'mpirun', '-n', str(ngpus), '-x', 'NCCL_DEBUG=INFO', sys.executable,
+        #     bert_pretrain_script], cwd=cwd)
```
Comment (thiagocrepaldi) [inline tools/ci_build/build.py:1213]:
```suggestion
```

File: tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-frontend-test-ci-pipeline.yml [hunk 3]
```diff
@@ -26,7 +26,12 @@ jobs:
     continueOnError: true
     condition: always()
 
-  # insert a python frontend test data preparation step here
+  # TODO: enable this script to avoid the need of having a copy of test data at agent VMs
+  # - script: |
+  #     orttraining/tools/ci_test/download_e2e_test_data.py \
+  #     --azure_blob_url https://onnxruntimetestdata.blob.core.windows.net/training/glue_data.zip
+  #     --target_dir /bert_data/hf_data/
+  #   displayName: 'Download training end-to-end test data'
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-frontend-test-ci-pipeline.yml:34]:
```suggestion
```

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by snnn** [comment]
Please reopen it is still needed

**Pull request closed without merging**